### PR TITLE
Fix toggle test and consolidate tracing step definitions

### LINKED ISSUE #5437: [CI] Test flake - Sidebar toggle
Sidebar toggle test fails when login flash page flashes. My guess is that when the login page flashes the navbar is unmounted where it previously existed causing the query to fail.

https://github.com/kiali/kiali/runs/8183621567?check_suite_focus=true

Screenshot:
Sidebar toggle -- Open the sidebar (failed)

### PR DESCRIPTION
**Pull request by nrfox**
Consolidates some step definitions shared across apps/services/workloads pages. Ensures that the toggle test waits for the page to load before beginning.

Fixes #5437

### CODE REVIEW

File: frontend/cypress/integration/common/kiali_sidebar_toggle.ts [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
+import { And, When, Then } from '@badeball/cypress-cucumber-preprocessor';
+import { ensureKialiFinishedLoading } from './transition';
 
 When('the sidebar is open', () => {
-  cy.get('#page-sidebar').should('be.visible').then(($sidebar) => {
-    if ($sidebar.attr('aria-hidden') === 'true') {
-      cy.get('#nav-toggle').click()
-    }
-  })
+  ensureKialiFinishedLoading();
```
Comment (nrfox) [inline frontend/cypress/integration/common/kiali_sidebar_toggle.ts:5]: This was the main change here to fix the toggle test.

File: frontend/cypress/integration/common/kiali_sidebar_toggle.ts [hunk 2]
```diff
@@ -1,28 +1,35 @@
...
   cy.get('#page-sidebar').should('not.be.visible');
 });
 
 When('the sidebar is closed', () => {
-  cy.get('#page-sidebar').should('be.visible').then(($sidebar) => {
-    if ($sidebar.attr('aria-hidden') === 'false') {
-      cy.get('#nav-toggle').click()
-    }
-  })
+  ensureKialiFinishedLoading();
```
Comment (nrfox) [inline frontend/cypress/integration/common/kiali_sidebar_toggle.ts:25]: Along with this

**Comment by nrfox** [comment]
I'm hoping this will fix the service details failures as well since adding `scrollBehavior: false` to the span click: https://github.com/kiali/kiali/pull/5473/files#diff-10df8ad2c83361548790a2c3f8bdccd623f51fd596da52f63600511fc14ed822R39. Normally before cypress interacts with an element it'll scroll to the element first and then click on it. My theory is that this scroll behavior was interfering the Kiali's own `RenderComponentScroll` behavior and would cause misclicks closing the opened menu by clicking off of it.

Example of failure: https://github.com/kiali/kiali/actions/runs/3106521234/jobs/5033643168

**Review by israel-hdez — APPROVED** [review]
Apparently, it needs a rebase.

It works OK here, so I'm approving.

File: frontend/src/app/AuthenticationController.tsx [hunk 3]
```diff
@@ -334,7 +334,8 @@ export class AuthenticationController extends React.Component<
       document.documentElement.className = isKiosk ? 'kiosk' : '';
       store.dispatch(GlobalActions.setKiosk(getKioskMode()));
     }
-  };R
+  };
+  R;
```
[the remark this answers is not in the document]
Comment (nrfox) [reply frontend/src/app/AuthenticationController.tsx:338]: haha I agree! It's not being used anywhere so my guess is that it was a typo that slipped through. I'll remove it.

[the remark this answers is not in the document]
Comment (israel-hdez) [reply frontend/src/app/AuthenticationController.tsx]: It may be a contribution from my daughter. It needs to go away :-)

**Review by israel-hdez — APPROVED** [review]
Nice to see CI green at first pass.

**Pull request merged**